app/controllers/etds_controller.rb

```diff
@@ -0,0 +1,163 @@
+    # SignaturePageService.regenerate_signature_page(submission)
+    render_created
+  end
+
+  private
+
+  def dissertation_id
+    etd_params.expect(:dissertationid)
+  end
+
+  def etd_params
+    params.expect(DISSERTATION: [:dissertationid, :title, :type, :readerapproval, :readercomment, :regapproval,
+                                 :regcomment, :documentaccess, :schoolname, :degreeconfyr, :univid, :sunetid,
+                                 :readeractiondttm, :regactiondttm, :degree, :name, :vpname, :career, :program,
+                                 :plan,
+                                 { sub: [%i[deadline]] },
+                                 { subplan: [%i[code __content__]],
+                                   reader: [%i[sunetid name_prefix prefix name suffix type
+                                               univid readerrole finalreader]] }])
+  end
+
+  def title
+    etd_params[:title].gsub(/\s+/, ' ').strip
+  end
+
+  def degree
+    return 'Ph.D.' if PHD_REGEX.match?(etd_params[:degree])
+    return 'Engineering' if ENG_REGEX.match?(etd_params[:degree])
+
+    etd_params[:degree]
+  end
+
+  def peoplesoft_actions
+    peoplesoft_service.new_reader_action(readers:, reader_action_attributes:)
+    peoplesoft_service.new_registrar_action(registrar_action_attributes:)
+  end
+
+  def peoplesoft_service
+    @peoplesoft_service ||= PeoplesoftService.new(submission:)
+  end
+
+  def readers
+    Reader.sorted_list(etd_params.expect(reader: [%i[sunetid name_prefix prefix name suffix type
+                                                     univid readerrole finalreader]]))
+  end
+
+  def reader_action_attributes
+    {
+      readerapproval: etd_params[:readerapproval],
+      readercomment: etd_params[:readercomment],
+      last_reader_action_at: etd_params[:readeractiondttm]&.in_time_zone(Settings.peoplesoft_timezone)
+    }
+  end
+
+  def registrar_action_attributes
+    {
+      regapproval: etd_params[:regapproval],
+      regcomment: etd_params[:regcomment],
+      last_registrar_action_at: etd_params[:regactiondttm]&.in_time_zone(Settings.peoplesoft_timezone)
+    }
+  end
+
+  def render_bad_request
+    render status: :bad_request, html: invalid_xml_message
+  end
+
+  def render_created
+    render status: :created, html: "#{submission.druid} #{message}"
+  end
+
+  # Authentication based on a allow list of server names and IP addresses
+  def request_from_authorized_origin?
+    begin
+      remote_hostname = Resolv.getname(request.remote_addr).downcase
+      return true if Settings.ps_servers.include?(remote_hostname)
+    rescue Resolv::ResolvError
+      return true if Settings.ps_ips.include?(request.remote_addr)
+    end
+    false
+  end
+
+  def authenticate
+    http_basic_authenticate_or_request_with(
+      name: Settings.dlss_admin,
+      password: Settings.dlss_admin_pw,
+      realm: 'Application',
+      message: 'You are unauthorized to perform this action'
+    )
```

Comment:
We need this here for our integration tests, so I wonder if we should short-circuit if the HB env is anything other than `qa` and `stage`?

Reply:
I'd be happy to handle that in follow up ;-)